backend/src/email/routes/tests/email-transactional.routes.test.ts

```diff
@@ -20,7 +39,9 @@ beforeEach(async () => {
     email: `user_${userId}@agency.gov.sg`,
   } as User)
   apiKey = await user.regenerateAndSaveApiKey()
-  userId += 1
+  mockSendEmail = jest
```

Comment:
Please don't move this here. It's over-generalisation. For tests, the default approach should be sharing as little as possible. Test cases should be easily readable and understandable in isolation

Reply:
You need a very very very very good reason for generalisation in test cases

Reply:
Fair enough! I was trying to generalise the code because I saw a lot of repetition, but I can totally see this particular `mockSendEmail` as going one step too far.

Reply:
Oh I guess it's not in the sense that some of the test case codepath would've terminated before hitting the `sendEmail` part? But otherwise the tests all assume that `sendEmail` would be successful?

Reply:
> Oh I guess it's not in the sense that some of the test case codepath would've terminated before hitting the `sendEmail` part? But otherwise the tests all assume that `sendEmail` would be successful?

That's assumption is not true (or at least not for long). We dont have it here yet but we might need to handle errors from `sendEmail` and write tests for that

Reply:
As I'm adding this back into each test, I think I see your point about overgeneralisation.

Basically, for the tests that fail even before reaching the `sendEmail` stage, there is no need to mock their resolved value, so mocking resolved value for all tests is an overkill!